### PR TITLE
Update test matrix

### PR DESCRIPTION
**Pull request by robsdedude**
- Use publicly available 4.3 images
 - Update test matrix to be more comprehensive
 - Run suitable integration tests also on cluster configurations

Backport of https://github.com/neo4j-drivers/testkit/pull/241

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -45,75 +45,84 @@
...
+    configurations = [
+        generate_config(version_, enterprise_, cluster_, scheme_, stress_test_)
+        for (version_, enterprise_, cluster_, scheme_, stress_test_) in (
+            # LTS version
+            # 3.5 servers only support routing scheme if configured as cluster
+            ("3.5",    False,       False,    "bolt",   0),
+            ("3.5",    True,        False,    "bolt",   0),
+            ("3.5",    True,        True,     "neo4j", 60),
+            # not officially supported versions
+            ("4.0",    True,        False,    "neo4j",  0),
```
Comment (fbiville) [inline main.py:98]: 4.0 has reached its end of life on July 14 (sorry I did not catch this earlier)
Shall we get rid of 4.0 altogether?

File: tests/neo4j/suites.py [hunk 2]
```diff
@@ -35,51 +35,24 @@
 """
 Suite for Neo4j 4.0
```
Comment (fbiville) [inline tests/neo4j/suites.py:36]: same as my question above: should we drop this?

**Review by fbiville — APPROVED** [review]

File: main.py (hunk 1 above)
Comment (robsdedude) [reply main.py:98]: I'd like to get the opinion of the other team members on this. I feel like it would be ok to drop it.

Comment (michael-simons) [reply main.py:98]: I would be fine with it, I think the differences in Cypher and Indizes are something that mostly effects things like OGM. 
3.5 we need to keep anyway, right?

Comment (fbiville) [reply main.py:98]: Yep, 3.5 EOL is November 28, 2021

**Pull request merged**